packages/react/src/components/header/components/headerActionBarItem/HeaderActionBarItemWithDropdown.tsx

```diff
@@ -0,0 +1,130 @@
+  /**
+   * Additional classname for the icon.
+   */
+  iconClassName?: string;
+  /**
+   * Additional classname for the dropdown element.
+   */
+  dropdownClassName?: string;
+  /**
+   * Label for the action bar item.
+   */
+  label: string | JSX.Element;
+  /**
+   * Label for the action bar item when dropdown is open.
+   */
+  closeLabel?: string | JSX.Element;
+  /**
+   * Icon for the action bar item.
+   */
+  icon: ElementType;
+  /**
+   * Icon for the action bar item when dropdown is open.
+   */
+  closeIcon?: ElementType;
+}> &
+  React.ComponentPropsWithoutRef<'div'>;
+
+export const HeaderActionBarItemWithDropdown = (properties: HeaderActionBarItemWithDropdownProps) => {
+  const {
+    id,
+    children,
+    label,
+    fullWidth,
+    className: classNameProp,
+    iconClassName: iconClassNameProp,
+    dropdownClassName: dropdownClassNameProp,
+    closeLabel,
+    icon,
+    closeIcon = IconCross,
+    ...props
+  } = properties;
+  const dropdownContentElementRef = useRef<HTMLElement>(null);
+  const containerElementRef = useRef<HTMLDivElement>(null);
+  const [visible, setDisplayProperty] = useState(false);
+  const [hasContent, setHasContent] = useState(false);
+
+  const getContainer = () => containerElementRef.current as HTMLDivElement;
+
+  const handleButtonClick = () => {
+    setDisplayProperty(!visible);
+  };
+
+  const handleDocumentClick = (event) => {
+    const container = getContainer();
+    const eventTargetNode = event.target;
+    if (!container.contains(eventTargetNode)) {
+      setDisplayProperty(false);
+    }
+  };
+
+  const handleBlur = (event) => {
+    const container = getContainer();
+    const eventTargetNode = event.relatedTarget;
+    if (!container.contains(eventTargetNode)) {
+      setDisplayProperty(false);
+    }
+  };
+
+  // Set event listener only when dropdown open
+  useEffect(() => {
+    if (visible) document.addEventListener('click', handleDocumentClick);
+    return () => document.removeEventListener('click', handleDocumentClick);
+  }, [containerElementRef.current, visible]);
+
+  // Hide the component if there is no content
+  useEffect(() => {
+    setHasContent(dropdownContentElementRef.current?.childNodes.length !== 0);
+  }, [children]);
+
+  const iconLabel = visible ? closeLabel : label;
+  const iconClass = visible ? closeIcon : icon;
+  const visibilityClasses = {
+    visible,
+    [classes.visible]: visible,
+    [classes.hasContent]: hasContent,
+    [classes.fullWidth]: fullWidth,
+  };
+  const className = classNames(classes.container, classNameProp, visibilityClasses);
+  const iconClassName = classNames(classes.icon, iconClassNameProp);
+  const dropdownClassName = classNames(classes.dropdown, dropdownClassNameProp, visibilityClasses);
+
+  /* eslint-disable jsx-a11y/no-noninteractive-tabindex */
+  return (
+    <div {...props} id={id} className={className} ref={containerElementRef} aria-haspopup onBlur={handleBlur}>
+      <HeaderActionBarItem
```

Comment:
Axe accessibility -plugin shows critical error with this element: the rendered button has no text nor aria-label.

https://dequeuniversity.com/rules/axe/4.7/button-name?application=AxeChrome